src/nvim/lua/executor.c

```diff
@@ -2076,3 +2076,34 @@ int nlua_do_ucmd(ucmd_T *cmd, exarg_T *eap, bool preview)
 
   return retv;
 }
+
+/// String representation of a Lua function reference
+///
+/// @return Allocated string
+char *nlua_funcref_str(LuaRef ref)
+{
+  lua_State *const lstate = global_lstate;
+  StringBuilder str = KV_INITIAL_VALUE;
+  kv_resize(str, 16);
+
+  if (!lua_checkstack(lstate, 1)) {
+    goto plain;
+  }
+  nlua_pushref(lstate, ref);
+  if (!lua_isfunction(lstate, -1)) {
+    lua_pop(lstate, 1);
+    goto plain;
+  }
+
+  lua_Debug ar;
```

Comment:
Do we know if this has any impact on performance? The Lua reference manual specifically says that the debug functions are slow.

Reply:
With mappings and autocommands this happens when printing them out, so shouldn't have any impact on performance at all.

With commands it happens when you define them though, so good point. I think the code there has to be refactored in general, because commands don't even have the `desc` field, the description is kept in rhs instead.

Reply:
To answer my own question, the implementation of `lua_getinfo` is LuaJIT looks quite lean: https://github.com/LuaJIT/LuaJIT/blob/7306ba78d62b55a64f25231df1d5697345a5572e/src/lj_debug.c

Reply:
Now the string representation for commands is rendered when you print them, it will no longer impact performance of defining commands.